### PR TITLE
Fix various vulnerabilities.

### LINKED ISSUE #4620: MSVC VS2022 compilation (linking) error
Started appearing today:
```
FAILED: binrz/rz-hash/rz-hash.exe 
"link"  /MACHINE:x64 /OUT:binrz/rz-hash/rz-hash.exe binrz/rz-hash/rz-hash.exe.p/rz-hash.c.obj "/release" "/nologo" "/OPT:REF" "librz/main/rz_main.lib" "librz/util/rz_util.lib" "librz/demangler/rz_demangler.lib" "librz/magic/rz_magic.lib" "librz/socket/rz_socket.lib" "librz/flag/rz_flag.lib" "librz/cons/rz_cons.lib" "librz/hash/rz_hash.lib" "librz/crypto/rz_crypto.lib" "librz/il/rz_il.lib" "librz/reg/rz_reg.lib" "librz/io/rz_io.lib" "librz/bp/rz_bp.lib" "librz/syscall/rz_syscall.lib" "librz/arch/rz_arch.lib" "librz/search/rz_search.lib" "librz/config/rz_config.lib" "librz/diff/rz_diff.lib" "librz/bin/rz_bin.lib" "librz/type/rz_type.lib" "librz/egg/rz_egg.lib" "librz/debug/rz_debug.lib" "librz/sign/rz_sign.lib" "librz/core/rz_core.lib" "librz/lang/rz_lang.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "advapi32.lib" "bcrypt.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
rz_arch.lib(isa_gnu_xtensa_xtensa-dis.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
rz_arch.lib(isa_gnu_xtensa_elf32-xtensa.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
   Creating library binrz\rz-hash\rz-hash.lib and object binrz\rz-hash\rz-hash.exp
binrz\rz-hash\rz-hash.exe : fatal error LNK1169: one or more multiply defined symbols found
[2002/2269] Linking target binrz/rz-run/rz-run.exe
FAILED: binrz/rz-run/rz-run.exe 
"link"  /MACHINE:x64 /OUT:binrz/rz-run/rz-run.exe binrz/rz-run/rz-run.exe.p/rz-run.c.obj "/release" "/nologo" "/OPT:REF" "librz/main/rz_main.lib" "librz/util/rz_util.lib" "librz/demangler/rz_demangler.lib" "librz/magic/rz_magic.lib" "librz/socket/rz_socket.lib" "librz/flag/rz_flag.lib" "librz/cons/rz_cons.lib" "librz/hash/rz_hash.lib" "librz/crypto/rz_crypto.lib" "librz/il/rz_il.lib" "librz/reg/rz_reg.lib" "librz/io/rz_io.lib" "librz/bp/rz_bp.lib" "librz/syscall/rz_syscall.lib" "librz/arch/rz_arch.lib" "librz/search/rz_search.lib" "librz/config/rz_config.lib" "librz/diff/rz_diff.lib" "librz/bin/rz_bin.lib" "librz/type/rz_type.lib" "librz/egg/rz_egg.lib" "librz/debug/rz_debug.lib" "librz/sign/rz_sign.lib" "librz/core/rz_core.lib" "librz/lang/rz_lang.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "advapi32.lib" "bcrypt.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
rz_arch.lib(isa_gnu_xtensa_xtensa-dis.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
rz_arch.lib(isa_gnu_xtensa_elf32-xtensa.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
   Creating library binrz\rz-run\rz-run.lib and object binrz\rz-run\rz-run.exp
binrz\rz-run\rz-run.exe : fatal error LNK1169: one or more multiply defined symbols found
[2003/2269] Linking target binrz/rz-asm/rz-asm.exe
FAILED: binrz/rz-asm/rz-asm.exe 
"link"  /MACHINE:x64 /OUT:binrz/rz-asm/rz-asm.exe binrz/rz-asm/rz-asm.exe.p/rz-asm.c.obj "/release" "/nologo" "/OPT:REF" "librz/main/rz_main.lib" "librz/util/rz_util.lib" "librz/demangler/rz_demangler.lib" "librz/magic/rz_magic.lib" "librz/socket/rz_socket.lib" "librz/flag/rz_flag.lib" "librz/cons/rz_cons.lib" "librz/hash/rz_hash.lib" "librz/crypto/rz_crypto.lib" "librz/il/rz_il.lib" "librz/reg/rz_reg.lib" "librz/io/rz_io.lib" "librz/bp/rz_bp.lib" "librz/syscall/rz_syscall.lib" "librz/arch/rz_arch.lib" "librz/search/rz_search.lib" "librz/config/rz_config.lib" "librz/diff/rz_diff.lib" "librz/bin/rz_bin.lib" "librz/type/rz_type.lib" "librz/egg/rz_egg.lib" "librz/debug/rz_debug.lib" "librz/sign/rz_sign.lib" "librz/core/rz_core.lib" "librz/lang/rz_lang.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "advapi32.lib" "bcrypt.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
rz_arch.lib(isa_gnu_xtensa_xtensa-dis.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
rz_arch.lib(isa_gnu_xtensa_elf32-xtensa.c.obj) : error LNK2005: __ucrt_int_to_float already defined in rz_arch.lib(isa_gnu_hppa_hppa-dis.c.obj)
   Creating library binrz\rz-asm\rz-asm.lib and object binrz\rz-asm\rz-asm.exp
binrz\rz-asm\rz-asm.exe : fatal error LNK1169: one or more multiply defined symbols found
ninja: build stopped: subcommand failed.
```
https://ci.appveyor.com/project/rizinorg/rizin/builds/50551826/job/oi6wfxpfwawdn8ir

### PR DESCRIPTION
**Pull request by wargio**
# DO NOT SQUASH
 

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

These vulns were found in r2 and i patched them manually.

I have backported also a PR which improves the FP rate when searching for x509 certificates.

Fix https://github.com/rizinorg/rizin/issues/4620

### CODE REVIEW

**Review by XVilka — APPROVED** [review]
Why draft? It looks good.

**Comment by wargio** [comment]
> Why draft? It looks good.

misclick

File: librz/include/rz_types_base.h [hunk 1]
```diff
@@ -175,15 +175,6 @@ typedef struct _utX {
 #undef B
 #define B4(a, b, c, d) ((a << 12) | (b << 8) | (c << 4) | (d))
 
-/* portable non-c99 inf/nan types */
-#if !defined(INFINITY)
```
Comment (XVilka) [inline librz/include/rz_types_base.h]: It will break older MSVC/SDK, no?

Comment (wargio) [reply librz/include/rz_types_base.h]: i have reverted back.

**Pull request merged**